src/encoded/viewconfigs/matrix.py

```diff
@@ -128,14 +125,14 @@ def _validate_items(self, type_info):
         if msg:
             raise HTTPBadRequest(explanation=msg)
 
-    def _set_query_aggs(self, query):
+    def _set_query_aggs(self, query, search_result, matrix):
```

Comment:
Let try removing the matrix dependency here too.  Put this in a separate commit than the search_results change in the other comments.  

_set_query_aggs only needs a single list for 
```
field in matrix['x']['facets'] or
                field in matrix['y']['facets']
```
and these two things 
```
        x_grouping = matrix['x']['group_by']
        y_groupings = matrix['y']['group_by']
```
for _construct_xygroupings

Reply:
I forgot move this to a separate commit. Is ther separate commit important to you?